tests/test_sdk.py

```diff
@@ -424,3 +426,76 @@ def test_create_trace_with_id_and_generation():
     span = trace["observations"][0]
     assert span["name"] == "generation"
     assert span["traceId"] == trace["id"]
+
+
+def test_end_generation():
+    langfuse = Langfuse(os.environ.get("LF_PK"), os.environ.get("LF_SK"), os.environ.get("HOST"))
+    api_wrapper = LangfuseAPI(os.environ.get("LF_PK"), os.environ.get("LF_SK"), os.environ.get("HOST"))
+
+    timestamp = datetime.now()
+    generation = langfuse.generation(
+        InitialGeneration(
+            name="query-generation",
+            startTime=timestamp,
+            model="gpt-3.5-turbo",
+            modelParameters={"maxTokens": "1000", "temperature": "0.9"},
+            prompt=[
+                {"role": "system", "content": "You are a helpful assistant."},
+                {
+                    "role": "user",
+                    "content": "Please generate the start of a company documentation that contains the answer to the questinon: Write a summary of the Q3 OKR goals",
+                },
+            ],
+            completion="This document entails the OKR goals for ACME",
+            usage=Usage(promptTokens=50, completionTokens=49),
+            metadata={"interface": "whatsapp"},
+        )
+    )
+
+    time.sleep(5)
+    end_time = datetime.now()
+    generation.end()
+
+    langfuse.flush()
+
+    assert len(langfuse.task_manager.result_mapping) == 3
+
+    trace_id = langfuse.get_trace_id()
+
+    trace = api_wrapper.get_trace(trace_id)
+
+    span = trace["observations"][0]
+    iso_format_end_time = end_time.astimezone(pytz.UTC).strftime("%Y-%m-%dT%H:%M:%S.%f")[:-3] + "Z"
+    assert span["endTime"] == iso_format_end_time
```

Comment:
Do you think these type of tests work reliably as they base on time? Maybe we should just check that the endtime is not null?

Reply:
I thought on the platform time is shown in UTC time zone. But yes also checking if it's null or not is better. I'll make the change.